.github/workflows/build_and_tests.yml

```diff
@@ -0,0 +1,37 @@
+name: Build H2M-Launcher and run tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build-and-tests:
+    name: Build and run tests
+    runs-on: windows-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 8.x
+
+      - name: Build project
+        run: dotnet build H2M-Launcher.sln --configuration Debug
+
+      - name: Run tests
+        run: dotnet test H2M-Launcher.sln --no-build --verbosity normal
+
+      - name: Publish H2MLauncher.UI
```

Comment:
I think we can skip publish and artifact uploading for pull requests

Reply:
Done